pkg/types/types.go

```diff
@@ -185,70 +184,42 @@ const (
 	SequentialCoordination = "sequential"
 )
 
-type Phase string
-
-const (
-	// FetchPhase encompasses the entire process from start to end, involves the finished event
-	FetchPhase Phase = "fetch"
-	// IndexerPhase involves a candidates-found|failure
-	IndexerPhase Phase = "indexer"
-	// QueryPhase involves a connect, query-asked|failure
-	QueryPhase Phase = "query"
-	// RetrievalPhase involves the full data retrieval: connect, proposed, accepted, first-byte-received, success|failure
-	RetrievalPhase Phase = "retrieval"
-)
-
 type EventCode string
 
 const (
-	CandidatesFoundCode    EventCode = "candidates-found"
-	CandidatesFilteredCode EventCode = "candidates-filtered"
-	StartedCode            EventCode = "started"
-	ConnectedCode          EventCode = "connected"
-	QueryAskedCode         EventCode = "query-asked"
-	QueryAskedFilteredCode EventCode = "query-asked-filtered"
-	ProposedCode           EventCode = "proposed"
-	AcceptedCode           EventCode = "accepted"
-	FirstByteCode          EventCode = "first-byte-received"
-	FailedCode             EventCode = "failure"
-	SuccessCode            EventCode = "success"
-	FinishedCode           EventCode = "finished"
+	CandidatesFoundCode          EventCode = "candidates-found"
+	CandidatesFilteredCode       EventCode = "candidates-filtered"
+	StartedCode                  EventCode = "started"
+	StartedFetchCode             EventCode = "started-fetch"
+	StartedFindingCandidatesCode EventCode = "started-finding-candidates"
+	StartedRetrievalCode         EventCode = "started-retrieval"
+	ConnectedToSPCode            EventCode = "connected-to-sp"
```

Comment:
just a suggestion - I've been finding "provider" to be a more pleasant shorthand to "SP", I got sick of seeing variations of `sp` in code when I did the provider scoring stuff and did a bit of a pivot then